pkg/cli/add_vcluster_helm.go

```diff
@@ -33,22 +39,105 @@ func AddVClusterHelm(
 	vClusterName string,
 	log log.Logger,
 ) error {
-	// check if vCluster exists
-	vCluster, err := find.GetVCluster(ctx, globalFlags.Context, vClusterName, globalFlags.Namespace, log)
-	if err != nil {
-		return err
+	var vClusters []find.VCluster
+	if options.All {
+		log.Debugf("add vcluster called with --all flag")
+		kubeClientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{
+				CurrentContext: globalFlags.Context,
+			})
+		hostClusterRestConfig, err := kubeClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		hostKubeClient, err := kubernetes.NewForConfig(hostClusterRestConfig)
+		if err != nil {
+			return err
+		}
+		namespaces, err := hostKubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		log.Debugf("looking for vclusters in %d namespaces", len(namespaces.Items))
+		for _, ns := range namespaces.Items {
+			log.Infof("looking for a vcluster in %s namespace", ns.GetName())
+			vClustersInNamespace, err := find.ListVClusters(ctx, globalFlags.Context, "", ns.GetName(), log)
+			if err != nil {
+				return err
+			}
+			if len(vClustersInNamespace) == 0 {
+				log.Infof("no vClusters found in context %s and namespace %s", globalFlags.Context, ns.GetName())
+				continue
+			}
+			vClusters = append(vClusters, vClustersInNamespace...)
```

Comment:
I don't think you need to list all the namespaces and then iterate over them. I see we [pass an empty ns ](https://github.com/loft-sh/vcluster/blob/cf9a1a48a2b38594df2c4139bbb1d62f23f9bf69/pkg/cli/list_helm.go#L96) to `ListVClusters` elsewhere. That should just give you the vclusters in all namespaces.